### DISCUSSION TITLE
Pad sensitivity setting

### DISCUSSION THREAD

**Discussion by egasimus** [bangcorrupt/hacktribe#316] [Q&A] [answered]
A few months back, I ruined my E2's pads with some bad cleaning solution. (I'm dumb.) I think it might've dissolved part of the conductive coating of the thin plastic membrane between the rubber pads and the PCB. It sat half-assembled in a box for a while. :facepalm: 

Today, I revisited it.

* First, I gave the PCB and membrane a few alcohol baths and microfiber scrubs (with the kind of cloth you'd use to clean eyeglasses)
* Then, I used some [two-sided sticky tape between the rubber pads and the membrane](http://www.korgforums.com/forum/phpBB2/viewtopic.php?t=99616) to lift the pads (without unpeeling the side towards the membrane).

I'm happy to report that 14 of 16 pads (all but the leftmost two) now respond to pressure. Since the tape I used is ~1mm thick (unlike the 0.5mm suggested in the thread), I do have to press them quite hard though. At least I know it isn't ruined permanently! :smiling_face_with_three_hearts: 

So, that made me wonder: is it possible to add an option to global settings which sets the base pad sensitivity? Assuming there's a threshold voltage that needs to be crossed before the pad registers as pressed, and that's a value defined in the firmware, as opposed to determined by the I/O board :thinking: Have you been able to pinpoint such a threshold value in the pad scan routine in the decompiled firmware?

**Comment by bangcorrupt** [comment, marked as answer]
The pad voltage is scanned by the MCU on the panel board, which then sends messages to the CPU via UART.  There is a pad pressure test mode, it won't help you fix it but it can show the response of each pad.  Hold [Pattern Set] + [MFX Send] while powering up and it will show meters of the pressure on each pad. I think there are only 32 steps of resolution.  

In normal mode, when a pad is hit, the MCU sends a single message with velocity value, so I don't think a threshold can be set in the CPU firmware.  It may be possible to hack the MCU at some point, but I haven't done any work on it.  It has an SWD header for debugging and writing it's internal flash.  It's firmware can also be updated from the CPU.

Have you tried contacting Korg for spare parts?

**Comment by papapanchal40-max** [comment]
pad sensivity

**Reply by vanana-old-acc** [reply]
Man I couldn't agree more
